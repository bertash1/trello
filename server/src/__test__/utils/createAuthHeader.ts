import { generateTokens } from "../../service/token"
import User from "../../models/User"

export const createAuthHeader = async () => {
  const user = await User.findOne()
  if (user) {
    const { email, isActivated, _id } = user
    const tokens = generateTokens({ email, isActivated, _id })
    return `bearer ${tokens.accessToken}`
  }
}
