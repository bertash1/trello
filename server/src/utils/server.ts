import express from "express"
import cardRouter from "../routes/card"
import taskRouter from "../routes/task"
import userRouter from "../routes/user"
import boardRouter from "../routes/board"
import commentRouter from "../routes/comment"
import cors from "cors"
import morgan from "morgan"
import cookieParser from "cookie-parser"
import dotenv from "dotenv"
import errorMiddleware from "../middlewares/error-middleware"

dotenv.config()

const createServer = () => {
  const app = express()

  app.use(express.json())
  app.use(cookieParser())
  app.use(morgan("common"))
  app.use(cors())
  app.use("/api", userRouter)
  app.use("/api/card", cardRouter)
  app.use("/api/task", taskRouter)
  app.use("/api/board", boardRouter)
  app.use("/api/comment", commentRouter)

  app.use(errorMiddleware)

  return app
}

export default createServer
