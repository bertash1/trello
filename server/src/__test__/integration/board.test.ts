import supertest from "supertest"

import createServer from "../../utils/server"
import { createAuthHeader } from "../utils/createAuthHeader"

const app = createServer()

describe("Board", () => {
  test("should return all user boards", async () => {
    const accessToken = await createAuthHeader()

    if (accessToken) {
      await supertest(app)
        .get("/api/board/userBoards")
        .set("Authorization", accessToken)
        .then((res) => {
          expect(res.status).toBe(200)
          expect(res.body).toMatchSnapshot()
        })
    }
  })

  test("should handle request without access token and return a 401 status", async () => {
    const request = await supertest(app).get("/api/board/userBoards")

    expect(request.status).toBe(401)
  })
})
