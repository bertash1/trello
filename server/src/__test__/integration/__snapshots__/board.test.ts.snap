// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should return all user boards 1`] = `
Array [
  Object {
    "_id": "61b8b3b1bac41bce43873f0c",
    "title": "testBoard",
  },
]
`;
